#727 Integrate REST API with the Waku V2 node

LNSD opened the pull request:
Integrate the REST API server to the Waku V2 node. With this, the REST API MVP would be concluded.

- [x] Mount the REST server during the setup sequence
- [x] Add the corresponding configuration parameters to the node configuration
- [ ] Document the currently available API endpoints and configuration (RFC, operators' manual, etc.) 

This PR resolves partially #727

LNSD commented on waku/v2/node/config.nim:
```diff
@@ -263,6 +263,38 @@ type
+      defaultValue: 8546
+      name: "rest-port" }: uint16
+    
+    restAdmin* {.
+      desc: "Enable access to REST HTTP Admin API: true|false",
+      defaultValue: false
+      name: "rest-admin" }: bool
+    
+    restPrivate* {.
+      desc: "Enable access to REST HTTP Private API: true|false",
+      defaultValue: false
+      name: "rest-private" }: bool
```
These two APIs are not available yet. Added as placeholders

LNSD commented:
@jm-clius and @kaiserd you can find the node documentation rendered: https://github.com/status-im/nwaku/blob/32b7b64391f5647eeb9400077ef52209d8ea212b/docs/api/v2/node.md#http-rest-api

jm-clius commented:
Looks good! Feel free to mark as ready for review when it is. WDYT about splitting the node.md doc and having a separate doc for the rest api?

LNSD commented:
> Looks good! Feel free to mark as ready for review when it is. WDYT about splitting the node.md doc and having a separate doc for the rest api?

Sounds good to me. This is something I was thinking about but I do not know how do we want to structure the docs/operator manual.

LNSD commented:
I have split the REST API from the nim API markdown file. And added a couple more links to the REST API documentation.

I need to do a little research about how to add a nimble/makefile task that generates the full OpenAPI spec (merging all the specs) so we can run it manually now and at the CI later. And with that, we would be able to generate easily the REST API documentation. This work will be part of another PR.

kaiserd reviewed (approved):
:sparkle: LGTM :sparkle:

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :x: | 47b56fd9 | [#1](https://ci.status.im/job/nim-waku/job/prs/job/linux/job/PR-1022/1/) | 2022-06-28 17:49:31 | ~21 min | `linux` | [:page_facing_up:`log`](https://ci.status.im/job/nim-waku/job/prs/job/linux/job/PR-1022/1/consoleText) |

jm-clius commented on docs/api/v2/rest-api.md:
```diff
@@ -0,0 +1,55 @@
+## HTTP REST API
+
+As the JSON-RPC API, the HTTP REST API consist of a set of methods opearting on the Waku Node remotelly over HTTP.
```
```suggestion
Similar to the JSON-RPC API, the HTTP REST API consists of a set of methods operating on the Waku Node remotely over HTTP.
```

jm-clius commented on waku/v2/node/config.nim:
```diff
@@ -263,6 +263,38 @@ type
+
+    restPort* {.
+      desc: "Listening port of the REST HTTP server.",
+      defaultValue: 8546
```
@kaiserd makes a good point. People running several nodes locally often simply use `--ports-shift` in 1-port increments to easily avoid port clashes. It seems reasonable that some people may want to run both the JSON-RPC and HTTP REST APIs at once while we're migrating functionality from one to the other. Perhaps we can use some completely different port here?

jm-clius reviewed (approved):
LGTM! Can't wait to try it. :)
